com.unity.netcode.gameobjects/Tests/Runtime/NetworkVariableTests.cs

```diff
@@ -79,6 +81,12 @@ public void Awake()
             TheDictionary.OnDictionaryChanged += DictionaryChanged;
         }
 
+        public void OnDestroy()
```

Comment:
Horrible, again, I could use some help on how to avoid needing this to avoid leaks (or at least leak warnings!)

Reply:
since `NetworkSet` is still a class, we could use its dtor to dispose underlying data structure there.
but if/when we turn it into a struct, we'll need to call Dispose() on it explicitly.
it's generally a good practice to use MonoBehaviour's Awake/OnDestroy pairs for alloc/free.
I think these look good to me, except I'd alloc TheSet in the Awake method instead of relying on ctor. (future-proofing)

Reply:
I don't think we can use the dtor here because the Native Collection will complain, this is because dtors only happen when the GC happens which is not always when the object is specifically no longer referenced. So this is the proper location to Dispose the collection.

Reply:
There is no Awake, NetworkVariableBase is not a monobehaviour.  That is why (and *this* is the part I truly I don't like) this 'OnDestroy()' has to be called in the OnDestroy() method of NetworkVariableTests.

Maybe instead I need to NetworkBehaviour scan and clean up its NetworkVariable instances...

Reply:
oh, that makes sense.

on the second comment though, whatever creates/registers/initializes NetVar should clean it — in this case, sounds like NetBhv might be a good candidate for that since it registers NetVars and could clean them up in its OnDestroy().
I don't remember the code very well though, if we recreate (therefore down the line realloc) them on connect/spawn, we might pair spawn/despawn, if not, we might dealloc on OnDestroy().